### PR TITLE
Draft: Batching improvements for performance

### PR DESCRIPTION
**Pull request by rick-r3**
Bulk methods will be overloaded in ENT.

Also fixed a test that doesn't work on Mac so that it is ignored on that platform.

### CODE REVIEW

**Comment by rick-r3** [comment]
Will need rebasing

**Comment by colljos** [comment]
API changes look good to me.

File: core/src/main/kotlin/net/corda/core/node/services/IdentityService.kt [hunk 1]
```diff
@@ -116,6 +116,15 @@ interface IdentityService {
         }
     }
 
+    /**
+     * Bulk version of [wellKnownPartyFromAnonymous]. Every input [AbstractParty] is present in the returned map.
+     *
+     * From 4.11 onwards.
+     */
+    fun wellKnownPartiesFromAnonumous(parties: Iterable<AbstractParty>): Map<AbstractParty, Party?> {
```
Comment (adelel1) [inline core/src/main/kotlin/net/corda/core/node/services/IdentityService.kt]: typo "Anonumous"

File: core/src/test/kotlin/net/corda/core/crypto/CryptoUtilsTest.kt [hunk 2]
```diff
@@ -936,6 +940,7 @@ class CryptoUtilsTest {
     @Test(timeout=300_000)
 	fun `test default SecureRandom uses platformSecureRandom`() {
         Assume.assumeFalse(IS_OPENJ9) // See CORDA-4055
+        Assume.assumeFalse(IS_MAC)
```
Comment (adelel1) [inline core/src/test/kotlin/net/corda/core/crypto/CryptoUtilsTest.kt:943]: Updated FIX version for this issue as 4.12.

Comment (rick-r3) [reply core/src/test/kotlin/net/corda/core/crypto/CryptoUtilsTest.kt:943]: Do you mean 4.11 ?

File: core/src/main/kotlin/net/corda/core/node/services/IdentityService.kt (hunk 1 above)
Comment (rick-r3) [reply core/src/main/kotlin/net/corda/core/node/services/IdentityService.kt]: Good spot.  Fixed.

**Pull request closed without merging**